packages/costa/src/runtime.ts

```diff
@@ -190,31 +202,42 @@ export class CostaRuntime {
   }
   /**
    * fetch and check if the package name is valid.
-   * @param name the plugin package name.
+   * @param name the plugin package name or ReadStream for npm package tarball.
    * @param cwd the current working directory.
    */
-  async fetch(name: string, cwd?: string): Promise<PluginPackage> {
-    const source = this.getSource(name, cwd || process.cwd());
+  async fetch(name: string | Readable, cwd?: string): Promise<PluginPackage> {
```

Comment:
This looks a little bit confusing because the `name` is not consistent with a Readable. I will have 2 alternatives:

1. change the `name` to `source`, then a string is to represent a source id, and a `Readable` is to represent a source stream, and we will describe this in the document.
2. move the `Readable` to another new method like `fetchFromStream(src)`.

Reply:
> 2. move the `Readable` to another new method like `fetchFromStream(src)`.
+1